src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordEqualityContractProperty.cs

```diff
@@ -128,18 +130,26 @@ internal sealed class GetAccessorSymbol : SourcePropertyAccessorSymbol
                        propertyModifiers,
                        location,
                        syntax,
-                       hasBody: false,
+                       hasBody: true,
                        hasExpressionBody: false,
                        isIterator: false,
                        modifiers: new SyntaxTokenList(),
                        MethodKind.PropertyGet,
                        usesInit: false,
-                       isAutoPropertyAccessor: true,
+                       isAutoPropertyAccessor: false,
                        isNullableAnalysisEnabled: false,
                        diagnostics)
             {
             }
 
+            internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+            {
+                base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
+
+                var compilation = this.DeclaringCompilation;
+                AddSynthesizedAttribute(ref attributes, compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
```

Comment:
>AddSynthesizedAttribute [](http://example.com/codeflow?start=16&length=23)

consider adding ```Debug.Assert(WellKnownMembers.IsSynthesizedAttributeOptional(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));```

Reply:
I don't understand the value of this assertion tbh, but I'm adding it anyway.

Reply:
>I don't understand the value of this assertion tbh

It is Ok to not report an error if a synthesized attribute is optional and "bad" in one way or another.